File: apps/web/src/types/activity.ts
export type ActivityStatus = 'Planned' | 'InProgress' | 'Completed' | 'Blocked' | 'Cancelled'

export const ACTIVITY_STATUSES: ActivityStatus[] = [
  'Planned',
  'InProgress',
  'Completed',
  'Blocked',
  'Cancelled',
]

export interface Activity {
  id: string
  userId: string
  projectId: string | null
  title: string
  description: string | null
  status: ActivityStatus
  startedAt: string | null
  finishedAt: string | null
  durationMinutes: number | null
  selfScore: number | null
  selfScoreComment: string | null
  createdAt: string
  updatedAt: string | null
}

export interface CreateActivityRequest {
  projectId?: string | null
  title: string
  description?: string | null
  status: ActivityStatus
  startedAt?: string | null
  finishedAt?: string | null
  durationMinutes?: number | null
  selfScore?: number | null
  selfScoreComment?: string | null
}

export interface UpdateActivityRequest {
  projectId?: string | null
  title: string
  description?: string | null
  status: ActivityStatus
  startedAt?: string | null
  finishedAt?: string | null
  durationMinutes?: number | null
  selfScore?: number | null
  selfScoreComment?: string | null
}

File: apps/backend/Migue/Migue.Domain/Entities/Activity.cs
using Migue.Domain.Enums;

namespace Migue.Domain.Entities;

public record Activity : BaseEntity
{
    public Guid UserId { get; set; }
    public Guid? ProjectId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public ActivityStatus Status { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? FinishedAt { get; set; }
    public int? DurationMinutes { get; set; }
    public int? SelfScore { get; set; }
    public string? SelfScoreComment { get; set; }

    public Project? Project { get; set; }
}
